gantry/firmware/step_motor.cpp

```diff
@@ -0,0 +1,29 @@
+#include "motor-control/core/step_motor.hpp"
+
+#include "common/firmware/timer_interrupt.h"
+#include "motor-control/core/motor_messages.hpp"
+
+using namespace motor_messages;
+
+static auto handler_class = motor_handler::MotorInterruptHandler<
+    freertos_message_queue::FreeRTOSMessageQueue>();
+
+void step_motor() {
+    if (handler_class.can_step()) {
+        toggle_step_pin();
```

Comment:
worth noting that this will step half as fast as you think since it can toggle once per interrupt. the two common approaches to this are to have the pin go on and off in the same interrupt, which costs you time since there's a minimum time the step pin must be high to make the driver work; or to always set the step pin low at the beginning of the interrupt, then set it high again if necessary, which might have a similar timing problem. i think this is fine for now since we're not setting velocity, but something to keep in mind.

Reply:
Yep, we're aware. We'll be making tweaks in following PRs. The main goal of the fixup was to mainly add in the queue.